$OpenBSD: patch-src_java_base_unix_native_libnio_ch_FileDispatcherImpl_c,v 1.1 2020/02/26 02:17:12 kurt Exp $

Don't throw IOException if the close(2) failure was due to ECONNRESET

https://github.com/battleblow/openjdk-jdk11u/commit/e0adb3c7fef6777e23c3b746a2a9d3db50bfebb8.patch

Index: src/java.base/unix/native/libnio/ch/FileDispatcherImpl.c
--- src/java.base/unix/native/libnio/ch/FileDispatcherImpl.c.orig
+++ src/java.base/unix/native/libnio/ch/FileDispatcherImpl.c
@@ -291,7 +291,7 @@ Java_sun_nio_ch_FileDispatcherImpl_release0(JNIEnv *en
 static void closeFileDescriptor(JNIEnv *env, int fd) {
     if (fd != -1) {
         int result = close(fd);
-        if (result < 0)
+        if (result < 0 && errno != ECONNRESET)
             JNU_ThrowIOExceptionWithLastError(env, "Close failed");
     }
 }
